### PR TITLE
Add bearer token generator

### PR DESCRIPTION
**Pull request by rlewan**
This PR provides an implementation of `AuthTokenGenerator` which prepends the `Bearer ` string to generated tokens. The new generator is intended to be used as a decorator for existing generators.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '0.1.0'
+version '0.2.0'
```
Comment (geek-soft) [inline build.gradle]: I would love to bump it to 1.0.0, anything below 1.0.0 seems a bit less serious to me.

**Comment by rkondratowicz** [comment]
I'd rather just introduce a breaking change to existing generator tbh

File: src/main/java/uk/gov/hmcts/reform/authorisation/generators/CachedServiceAuthTokenGenerator.java [hunk 2]
```diff
@@ -1,22 +0,0 @@
...
-            final int ttlInSeconds
-    ) {
-        cachedSupplier = memoizeWithExpiration(authTokenGenerator::generate, ttlInSeconds, TimeUnit.SECONDS);
-    }
-
-    @Override
-    public String generate() {
-        return cachedSupplier.get();
-    }
-}
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/authorisation/generators/CachedServiceAuthTokenGenerator.java]: does this mean we can remove `com.google.guava` library from dependencies?

Comment (rlewan) [reply src/main/java/uk/gov/hmcts/reform/authorisation/generators/CachedServiceAuthTokenGenerator.java]: Yes, good catch 👍

File: build.gradle (hunk 1 above)
Comment (rkondratowicz) [reply build.gradle]: very early days for this lib, not need to rush imho.
pdf-client is already 4.0.0 :)

**Pull request merged**